node/src/components/consensus/highway_core/highway.rs

```diff
@@ -203,6 +203,9 @@ impl<C: Context> Highway<C> {
             Vertex::Endorsements(endorsements) => {
                 if self.state.has_vote(endorsements.vote()) {
                     None
+                } else if self.state.is_endorsed(endorsements.vote()) {
+                    // Vote is not endorsed yet. Request more endorsements.
+                    Some(Dependency::Endorsement(*endorsements.vote()))
```

Comment:
An endorsement can't depend on more endorsements for the same vote.

Reply:
Duh, of course. This should go under a `vote` :facepalm:

Reply:
Yes, once we have the LNC, some votes will depend on endorsements. :+1:

Reply:
Done.